ts/features/itwallet/presentation/details/components/ItwPresentationWalletUpgradeMDLDetailsBanner.tsx

```diff
@@ -0,0 +1,92 @@
+import { Banner } from "@pagopa/io-app-design-system";
+import { useCallback, useMemo } from "react";
+import { useFocusEffect, useRoute } from "@react-navigation/native";
+import I18n from "../../../../../i18n";
+import { useIONavigation } from "../../../../../navigation/params/AppParamsList";
+import { ITW_ROUTES } from "../../../navigation/routes";
+import { itwSetWalletUpgradeMDLDetailsBannerHidden } from "../../../common/store/actions/preferences";
+import { useIODispatch, useIOSelector } from "../../../../../store/hooks";
+import { itwShouldRenderWalletUpgradeMDLDetailsBannerSelector } from "../../../common/store/selectors";
+import {
+  trackITWalletBannerVisualized,
+  trackItWalletBannerClosure,
+  trackItWalletBannerTap
+} from "../../../analytics";
+import { ITW_SCREENVIEW_EVENTS } from "../../../analytics/enum";
+import { StoredCredential } from "../../../common/utils/itwTypesUtils";
+
+type Props = {
+  credential: StoredCredential;
+};
+
+/**
+ * Banner promoting IT Wallet upgrade in MDL details to enable
+ * driving license usage as identity document.
+ */
+export const ItwPresentationWalletUpgradeMDLDetailsBanner = ({
+  credential
+}: Props) => {
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+  const shouldRender = useIOSelector(state =>
```

Comment:
We are instantiating objects even if they will not be used. What do you think about wrapping hooks and functions with a component that renders only if `itwShouldRenderWalletUpgradeMDLDetailsBannerSelector` returns true?